src/core/linux/Wayland.zig

```diff
@@ -21,53 +21,92 @@ pub const c = @cImport({
     @cInclude("linux/input-event-codes.h");
 });
 
+var libwaylandclient: LibWaylandClient = undefined;
+
+export fn wl_proxy_add_listener(proxy: ?*c.struct_wl_proxy, implementation: [*c]?*const fn () callconv(.C) void, data: ?*anyopaque) c_int {
+    return @call(.always_tail, libwaylandclient.wl_proxy_add_listener, .{ proxy, implementation, data });
+}
+
+export fn wl_proxy_get_version(proxy: ?*c.struct_wl_proxy) u32 {
+    return @call(.always_tail, libwaylandclient.wl_proxy_get_version, .{proxy});
+}
+
+export fn wl_proxy_marshal_flags(proxy: ?*c.struct_wl_proxy, opcode: u32, interface: [*c]const c.struct_wl_interface, version: u32, flags: u32, ...) ?*c.struct_wl_proxy {
+    var arg_list: std.builtin.VaList = @cVaStart();
+    defer @cVaEnd(&arg_list);
+
+    return @call(.always_tail, libwaylandclient.wl_proxy_marshal_flags, .{ proxy, opcode, interface, version, flags, arg_list });
+}
+
+export fn wl_proxy_destroy(proxy: ?*c.struct_wl_proxy) void {
+    return @call(.always_tail, libwaylandclient.wl_proxy_destroy, .{proxy});
+}
+
 state: *Core,
+core: *Core,
 title: [:0]const u8,
-size: Core.Size,
-surface_descriptor: gpu.Surface.Descriptor,
-configured: bool,
-decoration: *c.zxdg_toplevel_decoration_v1,
```

Comment:
didn't need to save this in state, because it is being used in the same scope it is being set